internal/pkg/log/log.go

```diff
@@ -1,157 +1,50 @@
 package log
 
 import (
-	"fmt"
 	"io"
-	"strings"
 
-	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
-type WriteCloser struct {
-	level  zapcore.Level
-	logger *zap.SugaredLogger
-}
-
-// Write messages with the defined level to logger.
-func (w *WriteCloser) Write(p []byte) (n int, err error) {
-	lines := strings.TrimRight(string(p), "\n")
-	for _, line := range strings.Split(lines, "\n") {
-		msg := strings.TrimRight(line, "\n")
-		switch w.level {
-		case zapcore.DebugLevel:
-			w.logger.Debug(msg)
-		case zapcore.InfoLevel:
-			w.logger.Info(msg)
-		case zapcore.WarnLevel:
-			w.logger.Warn(msg)
-		default:
-			w.logger.Error(msg)
-		}
-	}
-	return len(p), nil
-}
-
-func (w *WriteCloser) Close() error {
-	return w.logger.Sync()
-}
-
-func (w *WriteCloser) WriteString(s string) (n int, err error) {
-	return w.Write([]byte(s))
-}
-
-func (w *WriteCloser) WriteNoErr(p []byte) {
-	if _, err := w.Write(p); err != nil {
-		panic(fmt.Errorf("cannot write: %w", err))
-	}
-}
-
-func (w *WriteCloser) WriteStringNoErr(s string) {
-	if _, err := w.WriteString(s); err != nil {
-		panic(fmt.Errorf("cannot write: %w", err))
-	}
-}
-
-func (w *WriteCloser) WriteStringNoErrIndent1(s string) {
-	w.WriteStringNoErrIndent(s, 1)
-}
-
-func (w *WriteCloser) WriteStringNoErrIndent(s string, indent int) {
-	w.WriteStringNoErr(strings.Repeat("  ", indent) + s)
-}
-
-func ToDebugWriter(l *zap.SugaredLogger) *WriteCloser {
-	return &WriteCloser{zapcore.DebugLevel, l}
-}
-
-func ToInfoWriter(l *zap.SugaredLogger) *WriteCloser {
-	return &WriteCloser{zapcore.InfoLevel, l}
-}
-
-func ToWarnWriter(l *zap.SugaredLogger) *WriteCloser {
-	return &WriteCloser{zapcore.WarnLevel, l}
-}
+const (
```

Comment:
Vytvoril som interface `Logger` a `DebugLogger`, ktory sa bude pouzivat v celej apke.
`DebugLogger` je mozne pouzit miesto `Logger`.

Ostatny kod bol rozdeleny do viacerych suborov.

Z diffu to nie je asi uplne vidiet, subor vyzera teraz takto.
```go
package log

import (
	"io"

	"go.uber.org/zap/zapcore"
)

const (
	DebugLevel = zapcore.DebugLevel
	InfoLevel  = zapcore.InfoLevel
	WarnLevel  = zapcore.WarnLevel
	ErrorLevel = zapcore.ErrorLevel
)

type Logger interface {
	baseLogger
	sugaredLogger
	toWriter
}

// DebugLogger returns logs as string in tests.
type DebugLogger interface {
	Logger
	String() string
	Truncate()
	ConnectTo(writer io.Writer)
}

type baseLogger interface {
	Debug(args ...interface{})
	Info(args ...interface{})
	Warn(args ...interface{})
	Error(args ...interface{})
	Sync() error
}

type sugaredLogger interface {
	Debugf(template string, args ...interface{})
	Infof(template string, args ...interface{})
	Warnf(template string, args ...interface{})
	Errorf(template string, args ...interface{})
}

type toWriter interface {
	DebugWriter() *LevelWriter
	InfoWriter() *LevelWriter
	WarnWriter() *LevelWriter
	ErrorWriter() *LevelWriter
}
```